src/util/pointer_offset_size.cpp

```diff
@@ -681,6 +682,27 @@ optionalt<exprt> get_subexpression_at_offset(
       }
     }
   }
+  else if(
+    object_descriptor_exprt(expr).root_object().id() == ID_union &&
+    source_type.id() == ID_union)
+  {
+    const union_typet &union_type = to_union_type(source_type);
+
+    for(const auto &component : union_type.components())
+    {
+      const auto m_size_bits = pointer_offset_bits(component.type(), ns);
+      if(!m_size_bits.has_value())
+        continue;
+
+      // if this member completely contains the target, recurse into it
+      if(offset_bytes * 8 + *target_size_bits <= *m_size_bits)
```

Comment:
`CHAR_BIT` or `config.ansi_c.char_width`  You don't want to annoy the author of https://github.com/diffblue/cbmc/pull/917/ ;-)

Reply:
The author of #917 surely will want to consistently replace all occurrances of "8," so I'll leave it to him ;-P

Reply:
OK.  Not sure he's going to be happy but I'll let you fight that fight!